kafka_consumer/datadog_checks/kafka_consumer/client/confluent_kafka_client.py

```diff
@@ -25,11 +27,37 @@ def create_kafka_admin_client(self):
     def get_consumer_offsets_dict(self):
         raise NotImplementedError
 
-    def get_highwater_offsets(self):
-        raise NotImplementedError
+    def get_highwater_offsets(self, consumer_offsets):
+        # TODO: Remove broker_requests_batch_size as config after kafka-python is removed
```

Comment:
To reviewer: For additional context, we [introduced a config option](https://github.com/DataDog/integrations-core/pull/7093) called `broker_requests_batch_size` after running into performance issues using the `kafka-python` library. Now that we are converting to the Confluent library, I don't think we will need to batch our calls to get the highwater metrics since the [performance of confluent-kafka-python should be better than `kafka-python`](https://towardsdatascience.com/3-libraries-you-should-know-to-master-apache-kafka-in-python-c95fdf8700f2).